src/Testcontainers/Builders/TestcontainersCloudEndpointAuthenticationProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿namespace DotNet.Testcontainers.Builders
+{
+  using System;
+  using System.Text.Json;
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Images;
+  using JetBrains.Annotations;
+
+  /// <inheritdoc cref="IDockerRegistryAuthenticationProvider" />
+  [PublicAPI]
+  internal sealed class TestcontainersHostEndpointAuthenticationProvider : DockerEndpointAuthenticationProvider, ICustomConfiguration
+  {
+    private readonly ICustomConfiguration _customConfiguration = new TestcontainersHostConfiguration();
+
+    private readonly Uri _dockerEngine;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TestcontainersHostEndpointAuthenticationProvider" /> class.
+    /// </summary>
+    public TestcontainersHostEndpointAuthenticationProvider()
+    {
+      _dockerEngine = GetDockerHost();
+    }
+
+    /// <inheritdoc />
+    public override bool IsApplicable()
+    {
+      return _dockerEngine != null && "tcp".Equals(_dockerEngine.Scheme, StringComparison.OrdinalIgnoreCase);
+    }
+
+    /// <inheritdoc />
+    public override IDockerEndpointAuthenticationConfiguration GetAuthConfig()
+    {
+      return new DockerEndpointAuthenticationConfiguration(_dockerEngine);
+    }
+
+    /// <inheritdoc />
+    public string GetDockerConfig()
+    {
+      return _customConfiguration.GetDockerConfig();
+    }
+
+    /// <inheritdoc />
+    public Uri GetDockerHost()
+    {
+      return _customConfiguration.GetDockerHost();
+    }
+
+    /// <inheritdoc />
+    public string GetDockerHostOverride()
+    {
+      return _customConfiguration.GetDockerHostOverride();
```

Comment:
I see `_customConfiguration` is an instance of `PropertiesFileConfiguration`, does this also take into account environment variable configurations?

Reply:
No, it does not. It purely relies on the properties file. I asked myself the same question, if it should. Until now, TC for .NET use either of both ways to configure TC.

Reply:
I don't fully understand the question @cristianrgreco. 

We effectively want `TestcontainersHostConfiguration` to only be configured through the `testcontainers.host` property in the property file, right? And this implementation follows this and hence is correct I assume.